cloudpickle/cloudpickle.py

```diff
@@ -1134,6 +1140,9 @@ def _make_skel_func(code, cell_count, base_globals=None):
     if base_globals is None:
         base_globals = {}
     elif isinstance(base_globals, str):
+        # Backward compatibility for cloudpickle from 0.5.4 to 0.6.1, in which
```

Comment:
I flagged this code as backward compatibility code. But we can also remove it and replace it with 
```python 
elif isinstance(base_globals, str):
    base_globals = {}
```
This matters for pickle strings generated by cloudpickle between 0.5.4 and 0.6.1, and read by newer versions.
 @ogrisel would love to have your opinion on this.

Reply:
0.5.4 to 0.7.0 no?

Reply:
I think we might want to get rid of `_DynamicModuleFuncGlobals` but we might need to ensure that pickles generated between 0.5.4 and 0.7.0 that hold dynamically defined functions are still readable with versions (even if we now enforce the new isolation semantics).